apps/server/src/modules/board/service/board-do.service.spec.ts

```diff
@@ -1,155 +1,137 @@
-import { createMock, DeepMocked } from '@golevelup/ts-jest';
-import { BadRequestException } from '@nestjs/common';
+import { EntityManager } from '@mikro-orm/mongodb';
 import { Test, TestingModule } from '@nestjs/testing';
-import { setupEntities } from '@shared/testing';
+import { MongoMemoryDatabaseModule } from '@shared/infra/database';
+import { cleanupCollections } from '@shared/testing';
 import { cardFactory, columnFactory, textElementFactory } from '@shared/testing/factory/domainobject';
-import { Logger } from '@src/core/logger';
-import { BoardDoRepo } from '../repo';
+import { BoardDoRepo, BoardNodeRepo } from '../repo';
 import { RecursiveDeleteVisitor } from '../repo/recursive-delete.vistor';
 import { BoardDoService } from './board-do.service';
 
 describe(BoardDoService.name, () => {
 	let module: TestingModule;
 	let service: BoardDoService;
-	let boardDoRepo: DeepMocked<BoardDoRepo>;
+	let boardDoRepo: BoardDoRepo;
+	let em: EntityManager;
 
 	beforeAll(async () => {
 		module = await Test.createTestingModule({
-			providers: [
-				BoardDoService,
-				{
-					provide: BoardDoRepo,
-					useValue: createMock<BoardDoRepo>(),
-				},
-				{
-					provide: RecursiveDeleteVisitor,
-					useValue: createMock<RecursiveDeleteVisitor>(),
-				},
-				{
-					provide: Logger,
-					useValue: createMock<Logger>(),
-				},
-			],
+			imports: [MongoMemoryDatabaseModule.forRoot()],
```

Comment:
I would prefer to have a better api test for the move (checking the result), and keeping these isolated.

Reply:
Agreed. I added API tests and implemented the service test as a unit test again.